src/SkynetServer/Network/ProtocolException.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SkynetServer.Network
+{
+    internal class ProtocolException : Exception
+    {
+        public ProtocolException() : this("Client violation of Skynet Protocol rules caused an exception") { }
```

Comment:
I think that 'caused an exception' can be left out, as that should be pretty obvious at this point